### PR TITLE
Add patch mode to Set-AzVMOperatingSystem

### PR DESCRIPTION
**Pull request by grizzlytheodore**
## Description

adding PatchMode to set-AzVMOperatingSystem cmdlet. Per workitem: https://msazure.visualstudio.com/One/_workitems/edit/7538346

From Swagger/SDK update, New-AzDiskEncryptionSet response code was changed from 201, 202. So this test had to be re-recorded to pass. Also the exception error message was updated so the Assert-ThrowContains line was updated.
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by wyunchi-ms** [comment]
@grizzlytheodore , some test cases failed. Could you please take a look?

**Comment by grizzlytheodore** [comment]
@wyunchi-ms @hyonholee could you review and approve?

File: src/Compute/Compute/VirtualMachine/Config/SetAzureVMOperatingSystemCommand.cs [hunk 1]
```diff
@@ -225,6 +228,25 @@ public class SetAzureVMOperatingSystemCommand : Microsoft.Azure.Commands.Resourc
...
+        [Parameter(
+            ParameterSetName = WindowsDisableVMAgentParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Choose one of the following settings: 'Manual', 'AutomaticByOS', or 'AutomaticByPlatform'")]
+        [Parameter(
+            ParameterSetName = WindowsDisableVMAgentWinRmHttpsParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Choose one of the following settings: 'Manual', 'AutomaticByOS', or 'AutomaticByPlatform'")]
+        [ValidateNotNullOrEmpty]
+        public string PatchMode { get; set; }
```
Comment (hyonholee) [inline src/Compute/Compute/VirtualMachine/Config/SetAzureVMOperatingSystemCommand.cs:248]: Add [PSArgumentCompleter("Manual", "AutomaticByOS", "AutomaticByPlatform")]

File: src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1 [hunk 2]
```diff
@@ -862,8 +862,8 @@ function Test-DiskEncryptionSet
         #$mockkey2 = $kek2.Id
 
         $subId = Get-SubscriptionIdFromResourceGroup $rgname;
-        $mockkey1 = "https://kv1psenctest.vault.azure.net:443/keys/kek1psenctest/21571e3773bb4e6495c2d314a3f5de8b";
-        $mockkey2 = "https://kv2psenctest.vault.azure.net:443/keys/kek1psenctest/d4bae3704edb4d4da592360a756cd278";
+        $mockkey1 = "https://kv1psenctest.vault.azure.net/keys/kek1psenctest/7b0ea2a977294b93aa599d15c96a4368";
+        $mockkey2 = "https://kv2psenctest.vault.azure.net/keys/kek1psenctest/03684334d612487aa1bd8c9fb5349178";
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:866]: had to create new recources and re-record. using different keys.

File: src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1 [hunk 3]
```diff
@@ -890,7 +890,8 @@ function Test-DiskEncryptionSet
         $tags = @{test1 = "testval1"; test2 = "testval2" };
         Assert-ThrowsContains { `
             Update-AzDiskEncryptionSet -ResourceGroupName $rgname -Name $encryptionName -KeyUrl $mockkey2 -SourceVaultId $mocksourcevault2 -Tag $tags; } `
-            "Key rotation in disk encryption set is not supported in this version."
+            "Please grant get, wrap and unwrap key permissions to disk encryption set 'encpsenctest'."
+        #    "Key rotation in disk encryption set is not supported in this version."
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:894]: seems like CRP has updated the exception output

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.cs [hunk 4]
```diff
@@ -283,5 +283,19 @@ public void TestLowPriorityVirtualMachine()
...
+        {
+            TestRunner.RunTestScript("Test-SetAzVMOperatingSystem");
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestSetAzVMOperatingSystemError()
+        {
+            TestRunner.RunTestScript("Test-SetAzVMOperatingSystemError");
+        }
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.cs:299]: add two tests to check -PatchMode behavior for the cmdlet. One to check default value, the other one to check when it should error out

**Comment by grizzlytheodore** [comment]
@hyonholee  made updates to your comment. please approve

**Comment by grizzlytheodore** [comment]
all test cases that the PMs have requested have been added. 

There are file changes for:
-SDK package reference to the lastest version
-adding 'patchMode' parameter to Set-AzVMOperatingSystem
-fixing DiskEncryptionSet test failure
     it was failing because the Swagger update changed the response for a call, and and exception error message seems to be changed in the CRP level so the Assert-ThrowContains line had to be updated to expect that new message

**Comment by grizzlytheodore** [comment]
@wyunchi-ms @hyonholee please review and approve!

**Comment by wyunchi-ms** [comment]
LGTM

**Comment by grizzlytheodore** [comment]
@wyunchi-ms please merge

**Pull request merged**